### PR TITLE
fetches loadable_libraries using trunk api

### PR DESCRIPTION
**Pull request by shahadarsh**
I'm doing more testing on this mainly around `priority` and will do a follow-up PR with any improvements.

### CODE REVIEW

File: tembo-cli/src/cmd/apply.rs [hunk 1]
```diff
@@ -762,17 +766,21 @@ pub fn get_rendered_dockerfile(
...
+    let mut shared_preload_libraries: Vec<Library> = Vec::new();
 
     if let Some(ps_config) = postgres_configs {
         for p_config in ps_config.clone().into_iter() {
             match p_config.name.as_str() {
                 "shared_preload_libraries" => {
-                    shared_preload_libraries.push(p_config.value.to_string());
+                    shared_preload_libraries.push(Library {
+                        name: p_config.value.to_string(),
+                        priority: MAX_INT32,
```
Comment (shahadarsh) [inline tembo-cli/src/cmd/apply.rs:782]: I don't think there is any way to get priority for these so setting them to `2147483647`.

Comment (shahadarsh) [reply tembo-cli/src/cmd/apply.rs:782]: This is when `shared_preload_libraries` is specified in stack, app or directly for an instance.

File: tembo-cli/src/cmd/apply.rs [hunk 2]
```diff
@@ -823,10 +834,17 @@ fn get_postgres_config(
         }
     }
 
-    let config = shared_preload_libraries
+    let final_lodable_libs = Runtime::new().unwrap().block_on(get_loadable_libraries(
+        shared_preload_libraries.clone(),
+        extensions,
+    ));
+
+    let config = final_lodable_libs
```
Comment (sjmiller609) [inline tembo-cli/src/cmd/apply.rs:842]: Typo “lodable”

File: tembo-cli/src/cmd/apply.rs [hunk 3]
```diff
@@ -823,10 +834,17 @@ fn get_postgres_config(
...
+        .unwrap()
```
Comment (sjmiller609) [inline tembo-cli/src/cmd/apply.rs:843]: I’m not sure is this unwrap definitely safe or does it need error handling?

File: tembo-cli/src/cmd/apply.rs [hunk 4]
```diff
@@ -838,6 +856,41 @@ fn get_postgres_config(
...
+        let trunk_projects_url =
+            "https://registry.pgtrunk.io/api/v1/trunk-projects?extension-name=";
+
+        for ext in extensions.iter() {
+            let response = reqwest::get(format!("{}{}", trunk_projects_url, ext.name))
+                .await?
+                .text()
+                .await?;
+
+            let mut trunk_projects: Vec<TrunkProject> = serde_json::from_str(&response).unwrap();
```
Comment (sjmiller609) [inline tembo-cli/src/cmd/apply.rs]: Unhandled error case here with unwrap

File: tembo-cli/src/cmd/apply.rs [hunk 5]
```diff
@@ -838,6 +856,41 @@ fn get_postgres_config(
...
 
+async fn get_loadable_libraries(
+    mut shared_preload_libraries: Vec<Library>,
+    maybe_extensions: Option<&Vec<ControllerExtension>>,
+) -> Result<Vec<Library>, anyhow::Error> {
+    if let Some(extensions) = maybe_extensions {
+        let trunk_projects_url =
+            "https://registry.pgtrunk.io/api/v1/trunk-projects?extension-name=";
+
+        for ext in extensions.iter() {
```
Comment (sjmiller609) [inline tembo-cli/src/cmd/apply.rs:867]: I’m not sure but I think this logic is not handling the case of other trunk projects with matching extension names. Some different trunk projects have extensions of the same name

**Review by sjmiller609 — CHANGES_REQUESTED** [review]

File: tembo-cli/src/cmd/apply.rs (hunk 5 above)
Comment (sjmiller609) [reply tembo-cli/src/cmd/apply.rs:867]: I’m saying when you look up by extension name, you might also get other unrelated trunk projects with extensions of the same name

File: tembo-cli/src/cmd/apply.rs [hunk 6]
```diff
@@ -838,6 +856,41 @@ fn get_postgres_config(
...
+            let response = reqwest::get(format!("{}{}", trunk_projects_url, ext.name))
+                .await?
+                .text()
+                .await?;
+
+            let trunk_projects: Vec<TrunkProject> = serde_json::from_str(&response).unwrap();
```
Comment (sjmiller609) [inline tembo-cli/src/cmd/apply.rs:873]: also I think this unwrap looks unsafe, if the response from the API does not fit the TrunkProject struct for some reason

File: tembo-cli/src/cmd/apply.rs (hunk 3 above)
Comment (shahadarsh) [reply tembo-cli/src/cmd/apply.rs]: Thanks. Added error handling.

File: tembo-cli/src/cmd/apply.rs (hunk 4 above)
Comment (shahadarsh) [reply tembo-cli/src/cmd/apply.rs]: Done.

File: tembo-cli/src/cmd/apply.rs (hunk 5 above)
Comment (shahadarsh) [reply tembo-cli/src/cmd/apply.rs:873]: Please check the latest commit to see if it makes sense to you.

**Review by sjmiller609 — APPROVED** [review]

**Pull request merged**